airflow/models/taskinstance.py

```diff
@@ -1665,6 +1665,33 @@ def render_templates(self, context: Optional[Context] = None) -> None:
 
         self.task.render_template_fields(context)
 
+    def render_k8s_pod_yaml(self) -> Optional[str]:
+        """Render k8s pod yaml"""
+        from kubernetes.client.api_client import ApiClient
+
+        from airflow.executors.kubernetes_executor import KubeConfig, create_pod_id
+        from airflow.kubernetes.pod_generator import PodGenerator
+        from airflow.settings import pod_mutation_hook
+
+        kube_config = KubeConfig()
+        pod = PodGenerator.construct_pod(
+            dag_id=self.dag_id,
+            task_id=self.task_id,
+            pod_id=create_pod_id(self.dag_id, self.task_id),
+            try_number=self.try_number,
+            kube_image=kube_config.kube_image,
+            date=self.execution_date,
+            command=self.command_as_list(),
+            pod_override_object=PodGenerator.from_obj(self.executor_config),
+            scheduler_job_id="worker-config",
+            namespace=kube_config.executor_namespace,
+            base_worker_pod=PodGenerator.deserialize_model_file(kube_config.pod_template_file),
+        )
+        pod_mutation_hook(pod)
+        api_client = ApiClient()
```

Comment:
Should we not be using Client from https://github.com/apache/airflow/blob/master/airflow/kubernetes/kube_client.py#L97-L101

Reply:
It probably doesn't matter here as we aren't making an requests with this one.